### PR TITLE
feat(api): append cooments of column and table for metadata api

### PR DESCRIPTION
**Pull request by grieve54706**
This pull request includes several changes aimed at improving the metadata handling for various database connections in the `ibis-server` project. The most important changes include removing unnecessary `json` imports, updating methods to use `to_dict` instead of `to_json`, adding connection initialization in constructors, and enhancing the SQL queries to fetch additional metadata.

### Codebase Simplification:
* Removed unnecessary `json` imports across multiple files

### Method Updates:
* Updated methods to use `to_dict` instead of `to_json` for converting SQL query results to dictionaries.

### Connection Initialization:
* Added connection initialization in the constructors for `ClickHouseMetadata`, `MSSQLMetadata`, `MySQLMetadata`, and `PostgresMetadata` classes.

### SQL Query Enhancements:
* Enhanced SQL queries to include additional metadata such as table and column comments.

### Factory Pattern:
* Refactored `MetadataFactory` to use a mapping dictionary for cleaner and more maintainable code.

These changes collectively enhance the maintainability, readability, and functionality of the metadata handling code in the `ibis-server` project.

### CODE REVIEW

File: ibis-server/app/model/metadata/trino.py [hunk 1]
```diff
@@ -16,35 +13,45 @@
...
+                    c.is_nullable,
+                    c.column_comment
+                FROM
+                    information_schema.tables AS t
+                INNER JOIN
+                    information_schema.columns AS c
+                    ON t.table_catalog = c.table_catalog
+                    AND t.table_schema = c.table_schema
+                    AND t.table_name = c.table_name
+                WHERE t.table_schema = '{schema}'
```
Comment (goldmedal) [inline ibis-server/app/model/metadata/trino.py:38]: I didn't try it but maybe we can merge it with the table comments query? Something like
```sql
SELECT
    t.table_catalog,
    t.table_schema,
    t.table_name,
    c.column_name,
    c.data_type,
    c.is_nullable,
    c.column_comment,
    tc.comment
FROM
    information_schema.tables AS t
INNER JOIN
    information_schema.columns AS c
    ON t.table_catalog = c.table_catalog
    AND t.table_schema = c.table_schema
    AND t.table_name = c.table_name
INNER JOIN
    system.metadata.table_comments AS tc
    ON t.table_schema = tc.schema_name
    AND t.table_name = tc.table_name
WHERE t.table_schema = '{schema}'
```

File: ibis-server/tests/routers/v2/connector/test_mssql.py [hunk 2]
```diff
@@ -80,12 +81,32 @@ def mssql(request) -> SqlServerContainer:
...
+        )
+        conn.execute(
+            text("""
+                EXEC sys.sp_addextendedproperty 
+                    @name = N'MS_Description', 
+                    @value = N'This is a comment', 
+                    @level0type = N'SCHEMA', @level0name = 'dbo',
+                    @level1type = N'TABLE',  @level1name = 'orders',
+                    @level2type = N'COLUMN', @level2name = 'o_comment';
+            """)
```
Comment (goldmedal) [inline ibis-server/tests/routers/v2/connector/test_mssql.py:102]: 👍

**Review by goldmedal — COMMENTED** [review]
Thanks @grieve54706 working on this.
It seems that this PR doesn't implement for BigQuery to get the comment of tables or columns. Is there any plan for it?

File: ibis-server/app/model/metadata/trino.py (hunk 1 above)
Comment (grieve54706) [reply ibis-server/app/model/metadata/trino.py:38]: No. Trino does not implement `getTableHandle` for SystemMetadata. It will fail when executing.

**Comment by grieve54706** [comment]
> Thanks @grieve54706 working on this. It seems that this PR doesn't implement for BigQuery to get the comment of tables or columns. Is there any plan for it?

The BigQuery has already implemented this feature.

File: ibis-server/app/model/metadata/factory.py [hunk 3]
```diff
@@ -1,49 +1,28 @@
...
+from app.model.metadata.snowflake import SnowflakeMetadata
 from app.model.metadata.trino import TrinoMetadata
 
+mapping = {
+    DataSource.bigquery: BigQueryMetadata,
+    DataSource.clickhouse: ClickHouseMetadata,
+    DataSource.mssql: MSSQLMetadata,
+    DataSource.mysql: MySQLMetadata,
+    DataSource.postgres: PostgresMetadata,
+    DataSource.snowflake: SnowflakeMetadata,
```
[the remark this answers is not in the document]
Comment (grieve54706) [reply ibis-server/app/model/metadata/factory.py:17]: Sure

File: ibis-server/tests/routers/v2/connector/test_trino.py [hunk 4]
```diff
@@ -97,7 +115,7 @@ def test_query(trino: TrinoContainer):
         "1996-01-02",
         "1_370",
         "2024-01-01 23:59:59.000000",
-        "2024-01-01 23:59:59.000000 UTC",
+        "2024-01-01 23:59:59.000000",
```
Comment (goldmedal) [inline ibis-server/tests/routers/v2/connector/test_trino.py:118]: We didn't change the MDL and input data. Why does this value change? 🤔

**Comment by goldmedal** [comment]
There are some conflicts here. @grieve54706

File: ibis-server/tests/routers/v2/connector/test_trino.py (hunk 4 above)
Comment (grieve54706) [reply ibis-server/tests/routers/v2/connector/test_trino.py:118]: We changed the expression before, but we did not run CI for trino and did not change the excepted (https://github.com/Canner/wren-engine/pull/782)

**Review by goldmedal — APPROVED** [review]
Thanks @grieve54706 overall LGTM.

**Pull request merged**